routing/route.hpp

```diff
@@ -32,6 +33,51 @@ namespace routing
 using SubrouteUid = uint64_t;
 SubrouteUid constexpr kInvalidSubrouteId = std::numeric_limits<uint64_t>::max();
 
+/// \brief The route is composed of one or several subroutes. Every subroute is composed of segments.
+/// For every Segment is kept some attributes in the structure SegmentInfo.
+class RouteSegment final
+{
+public:
+  RouteSegment(Segment const & segment, turns::TurnItem const & turn, Junction const & junction,
+               std::string const & street, double distFromBeginningMeters, double distFromBeginningMerc,
+               double timeFromBeginningS, traffic::SpeedGroup traffic)
+    : m_segment(segment)
+    , m_turn(turn)
+    , m_junction(junction)
+    , m_street(street)
+    , m_distFromBeginningMeters(distFromBeginningMeters)
+    , m_distFromBeginningMerc(distFromBeginningMerc)
+    , m_timeFromBeginningS(timeFromBeginningS)
+    , m_traffic(traffic)
+  {
+  }
+
+  Segment const & GetSegment() const { return m_segment; }
+  turns::TurnItem const & GetTurn() const { return m_turn; }
+  Junction const & GetJunction() const { return m_junction; }
+  std::string const GetStreet() const { return m_street; }
+  double GetDistFromBeginningMeters() const { return m_distFromBeginningMeters; }
+  double GetDistFromBeginningMerc() const { return m_distFromBeginningMerc; }
+  double GetTimeFromBeginningS() const { return m_timeFromBeginningS; }
+  traffic::SpeedGroup GetTraffic() const { return m_traffic; }
+
+private:
+  Segment m_segment;
+  /// Turn (maneuver) information for the turn next to the |m_segment| if any.
+  /// If not |m_turn::m_turn| is equal to TurnDirection::NoTurn.
+  turns::TurnItem m_turn;
+  Junction m_junction; ///< The front point of segment.
```

Comment:
The furthest point of the segment from the beginning of the route along the route.